framework/include/mesh/SpiralAnnularMesh.h

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef SPIRALANNULARMESH_H
+#define SPIRALANNULARMESH_H
+
+#include "MooseMesh.h"
+
+class SpiralAnnularMesh;
+
+template <>
+InputParameters validParams<SpiralAnnularMesh>();
+
+/**
+ * Mesh generated from parameters
+ */
+class SpiralAnnularMesh : public MooseMesh
+{
+public:
+  SpiralAnnularMesh(const InputParameters & parameters);
+  SpiralAnnularMesh(const SpiralAnnularMesh & /* other_mesh */) = default;
+
+  // No copy
+  SpiralAnnularMesh & operator=(const SpiralAnnularMesh & other_mesh) = delete;
+
+  virtual std::unique_ptr<MooseMesh> safeClone() const override;
+
+  virtual void buildMesh() override;
+
+protected:
+  /// Radius of the inner circle
+  const Real & _inner_radius;
+
+  /// Radius of the outer circle. Logically, it's bigger that inner_radius.
+  const Real & _outer_radius;
+
+  /// Factor to increase initial_delta_r for each ring.
+  /// For a uniform grid : radial_bias = 1.0
+  Real _radial_bias;
+
+  /// Number of nodes on each ring.
+  const unsigned int & _nodes_per_ring;
+
+  /// Generate mesh of TRI6 elements instead of TRI3 elements.
+  const bool & _second_order;
+
+  /// Number of rings.You can't specify
+  /// both the number of rings and the radial bias if you want to match
+  /// a specified outer radius exactly... you have to leave one of
+  /// those parameters free so that it can be determined.
+  unsigned int _num_rings;
+
+  /// The boundary id to use for the cylinder.
+  const boundary_id_type &_cylinder_bid, _exterior_bid;
```

Comment:
Oh, I meant to mention when I saw this line of code earlier: the way that reference variable declarations work means that only the first of these (`_cylinder_bid`) is actually a reference type. The second (`_exterior_bid`) is a normal `boundary_id_type`. I think in this case there's no reason to make them references at all, so simply dropping the ampersand should be sufficient
```
const boundary_id_type _cylinder_bid, _exterior_bid;
```